### PR TITLE
fix golint #21

### LINKED ISSUE #21: fix golint and make golint fail to circleci
# Description: 

- [ ] Fix current golint issues we have. See log here

- [ ]  Make golint fail in circleci if there an offences ( at moment `make check` will always succeed)

```make check
>>> Reformatting code
pkg/apis/kubic/v1beta1/register.go
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/cmd/registries-operator/main.go:1:1: package comment should not have leading space
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/cmd/registries-operator/main.go:42:1: comment on exported var Version should be of the form "Version ..."
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/cmd/registries-operator/main.go:44:5: exported var Build should have comment or be unexported
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/apis/kubic/v1beta1/registry_types.go:49:1: comment on exported type RegistryCertificateStatus should be of the form "RegistryCertificateStatus ..." (with optional leading article)
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/apis/kubic/v1beta1/registry_types.go:76:1: comment on exported method Registry.GetCertificateSecret should be of the form "GetCertificateSecret ..."
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/apis/kubic/v1beta1/registry_types.go:91:1: receiver name r should be consistent with previous receiver name registry for Registry
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/config/globals.go:21:2: comment on exported var DefaultPrefix should be of the form "DefaultPrefix ..."
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/config/globals.go:24:2: comment on exported var DefaultDeployNumReplicas should be of the form "DefaultDeployNumReplicas ..."
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/test/util.go:1:1: package comment should be of the form "Package test ..."
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/test/util.go:1:1: package comment should not have leading space
/home/dmaiocchi/go/src/github.com/kubic-project/registries-operator/pkg/util/strings.go:26:6: func SafeId should be SafeID
```

### PR DESCRIPTION
**Pull request by u5surf**
## What does this PR change?

#21 

## Documentation
- No documentation needed: because it is not feature and only fix linting
- [x] **DONE**

## Test coverage
- test make check in localhost
```
$git pull --rebase fork issue-21
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (9/9), done.
remote: Total 16 (delta 9), reused 9 (delta 9), pack-reused 7
Unpacking objects: 100% (16/16), done.
From https://github.com/u5surf/registries-operator
 * branch            issue-21   -> FETCH_HEAD
 * [new branch]      issue-21   -> fork/issue-21
Updating 30a1301..e0094aa
Fast-forward
 cmd/registries-operator/main.go          | 6 +++++-
 pkg/apis/kubic/v1beta1/register.go       | 1 -
 pkg/apis/kubic/v1beta1/registry_types.go | 9 +++++----
 pkg/config/globals.go                    | 4 ++--
 pkg/test/util.go                         | 2 ++
 pkg/util/strings.go                      | 4 ++--
 6 files changed, 16 insertions(+), 10 deletions(-)
Current branch issue-21 is up to date.
$git log
commit e0094aaa69a91b0731dd3cffd4591c6bdd64cfc9 (HEAD -> issue-21, fork/issue-21)
commit e0094aaa69a91b0731dd3cffd4591c6bdd64cfc9 (HEAD -> issue-21, fork/issue-21
)
Author: u5surf <u5.horie@gmail.com>
Date:   Mon Nov 26 09:10:25 2018 +0900

    fix golint #21

commit 30a1301c0d8c22cf3c4cbf70af210362cc734fb2 (origin/master, origin/HEAD, master)
Merge: 534661b 9782e4e
Author: Rafael Fernández López <ereslibre@kde.org>
Date:   Fri Nov 23 16:59:58 2018 +0100

    Merge pull request #22 from kubic-project/fix-rbac

    Fix RBAC: Use cluster-manager role

commit 9782e4ec1612a5563ad5a60b13030d7288645163
Author: Pablo Chacin <pchacin@suse.com>
Date:   Fri Nov 23 16:23:01 2018 +0100

    Fix RBAC: Use cluster-manager role

    Use cluster-manager role instead of generated rbac while
$make check
>>> Reformatting code
go: finding github.com/golang/groupcache v0.0.0-20180924190550-6f2cf27854a4
go: finding github.com/google/gofuzz v0.0.0-20170612174753-24818f796faf
go: finding github.com/spf13/cobra v0.0.3
go: finding github.com/spf13/pflag v1.0.3
go: finding github.com/inconshreveable/mousetrap v1.0.0
go: finding github.com/go-logr/logr v0.1.0
go: finding github.com/stevvooe/resumable v0.0.0-20180830230917-22b14a53ba50
go: finding github.com/google/btree v0.0.0-20180813153112-4030bb1f1f0c
go: finding golang.org/x/crypto v0.0.0-20180222182404-49796115aa4b
go: finding go.uber.org/atomic v1.3.2
go: finding k8s.io/apimachinery v0.0.0-20180621070125-103fd098999d
go: finding github.com/modern-go/reflect2 v0.0.0-20180701023420-4b7aa43c6742
go: finding k8s.io/utils v0.0.0-20180918230422-cd34563cd63c
go: finding go.uber.org/multierr v1.1.0
go: finding github.com/ghodss/yaml v1.0.0
go: finding github.com/ugorji/go v1.1.1
go: finding k8s.io/api v0.0.0-20180712090710-2d6f90ab1293
go: finding golang.org/x/time v0.0.0-20180412165947-fbb02b2291d2
go: finding github.com/google/uuid v1.0.0
go: finding github.com/mattbaird/jsonpatch v0.0.0-20171005235357-81af80346b1a
go: finding github.com/pborman/uuid v0.0.0-20180906182336-adf5a7427709
go: finding github.com/docker/distribution v2.6.2+incompatible
go: finding github.com/go-logr/zapr v0.1.0
go: finding k8s.io/kube-openapi v0.0.0-20180928202339-9dfdf9be683f
go: finding golang.org/x/sys v0.0.0-20181005133103-4497e2df6f9e
go: finding k8s.io/apiextensions-apiserver v0.0.0-20180808065829-408db4a50408
go: finding golang.org/x/net v0.0.0-20181005035420-146acd28ed58
go: finding github.com/gregjones/httpcache v0.0.0-20180305231024-9cad4c3443a7
go: finding k8s.io/apiserver v0.0.0-20180808060109-1844acd6a035
go: finding go.uber.org/zap v1.9.1
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding github.com/json-iterator/go v1.1.5
go: finding golang.org/x/oauth2 v0.0.0-20181003184128-c57b0facaced
go: finding gopkg.in/inf.v0 v0.9.1
go: finding sigs.k8s.io/testing_frameworks v0.0.0-20180709092217-5818a3a284a1
go: finding github.com/peterbourgon/diskv v2.0.1+incompatible
go: finding sigs.k8s.io/controller-runtime v0.1.4
go: finding k8s.io/kubernetes v1.11.3
go: finding k8s.io/client-go v0.0.0-20180806134042-1f13a808da65
go: finding github.com/imdario/mergo v0.3.6
go: finding github.com/googleapis/gnostic v0.2.0
go: finding github.com/renstrom/dedent v1.0.0
go: downloading k8s.io/apimachinery v0.0.0-20180621070125-103fd098999d
go: downloading k8s.io/api v0.0.0-20180712090710-2d6f90ab1293
go: downloading sigs.k8s.io/controller-runtime v0.1.4
go: downloading k8s.io/kubernetes v1.11.3
go: downloading k8s.io/client-go v0.0.0-20180806134042-1f13a808da65
go: downloading github.com/google/gofuzz v0.0.0-20170612174753-24818f796faf
go: downloading k8s.io/apiserver v0.0.0-20180808060109-1844acd6a035
go: downloading github.com/spf13/pflag v1.0.3
go: downloading golang.org/x/oauth2 v0.0.0-20181003184128-c57b0facaced
go: downloading github.com/golang/groupcache v0.0.0-20180924190550-6f2cf27854a4
go: downloading github.com/spf13/cobra v0.0.3
go: downloading github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
go: downloading github.com/ghodss/yaml v1.0.0
go: downloading github.com/renstrom/dedent v1.0.0
go: downloading golang.org/x/net v0.0.0-20181005035420-146acd28ed58
go: downloading github.com/go-logr/zapr v0.1.0
go: downloading k8s.io/kube-openapi v0.0.0-20180928202339-9dfdf9be683f
go: downloading go.uber.org/zap v1.9.1
go: downloading github.com/go-logr/logr v0.1.0
go: downloading github.com/gregjones/httpcache v0.0.0-20180305231024-9cad4c3443a7
go: downloading cloud.google.com/go v0.30.0
go: downloading github.com/mattbaird/jsonpatch v0.0.0-20171005235357-81af80346b1a
go: downloading github.com/pborman/uuid v0.0.0-20180906182336-adf5a7427709
go: downloading github.com/peterbourgon/diskv v2.0.1+incompatible
go: downloading golang.org/x/time v0.0.0-20180412165947-fbb02b2291d2
go: downloading github.com/imdario/mergo v0.3.6
go: downloading github.com/json-iterator/go v1.1.5
go: downloading github.com/google/uuid v1.0.0
go: downloading github.com/google/btree v0.0.0-20180813153112-4030bb1f1f0c
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading go.uber.org/atomic v1.3.2
go: downloading github.com/googleapis/gnostic v0.2.0
go: downloading golang.org/x/crypto v0.0.0-20180222182404-49796115aa4b
go: downloading go.uber.org/multierr v1.1.0
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v0.0.0-20180701023420-4b7aa43c6742
go: downloading golang.org/x/sys v0.0.0-20181005133103-4497e2df6f9e
go: downloading k8s.io/utils v0.0.0-20180918230422-cd34563cd63c
go: downloading github.com/ugorji/go v1.1.1
>>> Reformatting code
```
- [x] **DONE**

## Links
(add here link to GitHub Issues)
Fixes #21 

- [x] **DONE**

### CODE REVIEW

**Comment by MalloZup** [comment]
Thx! i will check later and let u know :)

File: pkg/config/globals.go [hunk 1]
```diff
@@ -18,9 +18,9 @@
 package config
 
 var (
-	// A prefix for all the resources created by the operator
+	//DefaultPrefix for all the resources created by the operator
```
Comment (ereslibre) [inline pkg/config/globals.go]: Missing leading space in comment

File: pkg/apis/kubic/v1beta1/registry_types.go [hunk 2]
```diff
@@ -88,8 +88,9 @@ func (registry Registry) GetCertificateSecret(r client.Client) (*v1.Secret, erro
 
 }
 
-func (r Registry) String() string {
-	return fmt.Sprintf("%s", r.Spec.HostPort)
+// String returns Registry HOST:PORT formatted address
```
Comment (ereslibre) [inline pkg/apis/kubic/v1beta1/registry_types.go]: I think `Registry` here could be lowercase

File: pkg/test/util.go [hunk 3]
```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  *
  */
+
+// Package test is defined skip the specific test if we are not use running in integration testing mode
```
Comment (ereslibre) [inline pkg/test/util.go]: I would make this comment more generic, something like: `Package test contains specific test utilities used throughout the operator`

File: cmd/registries-operator/main.go [hunk 4]
```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  *
  */
+
+// Command main is a cli for this k8s operator
```
Comment (MalloZup) [inline cmd/registries-operator/main.go]: i think for `golint` we don't need to comment the package name , so we can remove this

Comment (u5surf) [reply cmd/registries-operator/main.go]: remove this but failed on local...
```
$golint cmd/registries-operator/main.go
cmd/registries-operator/main.go:1:1: package comment should not have leading space
```

Comment (u5surf) [reply cmd/registries-operator/main.go]: ok i understand all
ref: https://github.com/golang/lint/issues/22

Comment (u5surf) [reply cmd/registries-operator/main.go]: ok i understand all.
ref: https://github.com/golang/lint/issues/22

Comment (MalloZup) [reply cmd/registries-operator/main.go]:
```golang

 * limitations under the License. */

package main

``

**Comment by MalloZup** [comment]
@ereslibre  all you remarks were adressed, once you have time feel free to check and we can merge this.

thx for this contrib

**Pull request merged**